system/Traits/ConditionalTrait.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/**
+ * This file is part of CodeIgniter 4 framework.
+ *
+ * (c) CodeIgniter Foundation <admin@codeigniter.com>
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+namespace CodeIgniter\Traits;
+
+trait ConditionalTrait
+{
+    /**
+     * Only runs the query when $condition evaluates to true
+     *
+     * @param array|bool|float|int|object|resource|string|null $condition
```

Comment:
How about using `@template` types for the `$condition` so that when the trait is actually used, we can annotate the actual type on it and static analyzers can check it.

Reply:
Could you give an example? I don't see where a template would help because the value is passed through to the callable, which should have its own typed parameters.

Reply:
Actually not annotate, but let static analyzers like PHPStan analyse the condition's type.
This is a contrived example where template type may be useful.

https://phpstan.org/r/52e48ac1-6f8e-4b8d-9c51-f53918f4f64e

Reply:
Using the current typing (for the sake of discussion I used `mixed` instead of the long form):
https://phpstan.org/r/c6ed6719-28f3-4eed-acf9-3f50bb19065b

Reply:
Ah I see what you mean! Yes I think that would be a good idea. I'm always for more specific types 💪